boa_builtins/src/lib.rs

```diff
@@ -0,0 +1,181 @@
+use std::{
+    fmt::{self, Debug},
+    hash::{Hash, Hasher},
+};
+
+use bitflags::bitflags;
+use phf::PhfHash;
+use phf_shared::PhfBorrow;
+
+bitflags! {
+    /// This struct constains the property flags as described in the ECMAScript specification.
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+    pub struct Attribute: u8 {
+        /// The `Writable` attribute decides whether the value associated with the property can be changed or not, from its initial value.
+        const WRITABLE = 0b0000_0001;
+
+        /// If the property can be enumerated by a `for-in` loop.
+        const ENUMERABLE = 0b0000_0010;
+
+        /// If the property descriptor can be changed later.
+        const CONFIGURABLE = 0b0000_0100;
+    }
+}
+
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+pub struct StaticString {
+    index: u16,
+}
+
+impl StaticString {
+    #[inline]
+    pub fn index(self) -> u16 {
+        self.index
+    }
+}
```

Comment:
Maybe we should extract all our value types into a `boa_values` crate. That should allow us to use `JsString` and `JsSymbol`'s const methods directly, instead of defining this.